src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+import { AutomatedAnalysisScoreFilter } from './Filters/AutomatedAnalysisScoreFilter';
+
+interface AutomatedAnalysisCardProps {
+  pageTitle: string;
+}
+
+export const AutomatedAnalysisCard: React.FunctionComponent<AutomatedAnalysisCardProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const dispatch = useDispatch<StateDispatch>();
+
+  const [targetConnectURL, setTargetConnectURL] = React.useState('');
+  const [categorizedEvaluation, setCategorizedEvaluation] = React.useState<[string, RuleEvaluation[]][]>(
+    [] as [string, RuleEvaluation[]][]
+  );
+  const [filteredCategorizedEvaluation, setFilteredCategorizedEvaluation] = React.useState<
+    [string, RuleEvaluation[]][]
+  >([] as [string, RuleEvaluation[]][]);
+  const [isCardExpanded, setIsCardExpanded] = React.useState<boolean>(true);
+  const [isError, setIsError] = React.useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [reportStalenessTimer, setReportStalenessTimer] = React.useState<number>(0);
+  const [reportStalenessTimerUnits, setReportStalenessTimerUnits] = React.useState<string>('second');
+  const [reportTime, setReportTime] = React.useState<number>(0);
+  const [usingArchivedReport, setUsingArchivedReport] = React.useState<boolean>(false);
+  const [usingCachedReport, setUsingCachedReport] = React.useState<boolean>(false);
+  const [showNAScores, setShowNAScores] = React.useState<boolean>(false);
+
+  const SECOND_MILLIS = 1000;
+  const MINUTE_MILLIS = 60 * SECOND_MILLIS;
+  const HOUR_MILLIS = 60 * MINUTE_MILLIS;
+  const DAY_MILLIS = 24 * HOUR_MILLIS;
+
+  const targetAutomatedAnalysisFilters = useSelector((state: RootState) => {
+    const filters = state.automatedAnalysisFilters.state.targetFilters.filter(
+      (targetFilter: TargetAutomatedAnalysisFilters) => targetFilter.target === targetConnectURL
+    );
+    return filters.length > 0 ? filters[0].filters : emptyAutomatedAnalysisFilters;
+  }) as AutomatedAnalysisFiltersCategories;
+
+  const targetAutomatedAnalysisGlobalFilters = useSelector((state: RootState) => {
+    return state.automatedAnalysisFilters.state.globalFilters.filters;
+  }) as AutomatedAnalysisGlobalFiltersCategories;
+
+  const categorizeEvaluation = React.useCallback(
+    (arr: [string, RuleEvaluation][]) => {
+      const map = new Map<string, RuleEvaluation[]>();
+      arr.forEach(([_, evaluation]) => {
+        const topicValue = map.get(evaluation.topic);
+        if (topicValue === undefined) {
+          map.set(evaluation.topic, [evaluation]);
+        } else {
+          topicValue.push(evaluation);
+          topicValue.sort((a, b) => b.score - a.score || a.name.localeCompare(b.name));
+        }
+      });
+      setCategorizedEvaluation((Array.from(map) as [string, RuleEvaluation[]][]).sort());
+      setIsLoading(false);
```

Comment:
I think this method should only do the categorization task. Others should be handled outside.

Reply:
Do you mean that the setIsLoading and the setIsError shouldn;t be here?

Reply:
I think both?